crates/pop-cli/src/commands/up/rollup.rs

```diff
@@ -237,6 +341,70 @@ fn prompt_for_proxy_address(cli: &mut impl Cli) -> Result<String> {
 	Ok(format!("Id({address})"))
 }
 
+// Prompts the user the user to select a deployment provider or only register.
+fn prompt_provider(cli: &mut impl Cli) -> Result<Option<DeploymentProvider>> {
+	let mut predefined_action = cli.select("Select your deployment method:");
+	for action in DeploymentProvider::VARIANTS {
+		predefined_action =
+			predefined_action.item(Some(action.clone()), action.name(), action.description());
+	}
+	predefined_action = predefined_action.item(
+		None,
+		"Only Register in Relay Chain",
+		"Register the parachain in the relay chain without deploying",
+	);
+	Ok(predefined_action.interact()?)
+}
+
+// Prompts user to select a supported chain for deployment.
+fn prompt_supported_chain(cli: &mut impl Cli) -> Result<&SupportedChains> {
+	let mut chain_selected =
+		cli.select("Select a Relay Chain\n\nChoose from the supported relay chains:");
+	for chain in SupportedChains::VARIANTS {
+		chain_selected = chain_selected.item(chain, chain.to_string(), "");
+	}
+	Ok(chain_selected.interact()?)
+}
+
+// Prompts for an API key and attempts to read from environment first.
+fn prompt_api_key(env_var_name: &str, cli: &mut impl Cli) -> Result<String> {
+	if let Ok(api_key) = env::var(env_var_name) {
+		cli.info(format!("Using API key from environment variable ({env_var_name})."))?;
+		return Ok(api_key);
+	}
+	cli.warning(format!("No API key found for the environment variable `{env_var_name}`.\n{}", style(format!("Note: Consider setting this variable in your shell (e.g., `export {env_var_name}=...`) or system environment so you won’t be prompted each time.")).dim()))?;
+	let api_key = cli.password("Enter your API key:").interact()?;
+	Ok(api_key)
+}
+
+// Prompts the user to choose which template was used.
+fn prompt_template_used(cli: &mut impl Cli) -> Result<&str> {
```

Comment:
As explained in the PR description, `pop-cli` automatically determines the template used by reading the `basedOn` value from the chain spec. 
Since this functionality is introduced in this PR, older templates may not include this field. If `pop-cli `cannot automatically identify the template, it will prompt the user for it, as PDP requires this information.